mlflow/keras.py

```diff
@@ -754,31 +751,26 @@ def _run_and_log_function(self, original, args, kwargs, unlogged_params, callbac
 
             _log_early_stop_callback_metrics(early_stop_callback, history, metrics_logger)
 
-        if auto_end_run:
-            try_mlflow_log(mlflow.end_run)
-
         return history
 
-    def fit(self, *args, **kwargs):
-        original = gorilla.get_original_attribute(keras.Model, "fit")
+    def fit(original, self, *args, **kwargs):
         unlogged_params = ["self", "x", "y", "callbacks", "validation_data", "verbose"]
         return _run_and_log_function(self, original, args, kwargs, unlogged_params, 5)
 
-    def fit_generator(self, *args, **kwargs):
+    def fit_generator(original, *args, **kwargs):
         """
         NOTE: `fit_generator()` is deprecated in Keras >= 2.4.0 and simply wraps `fit()`.
         To avoid unintentional creation of nested MLflow runs caused by a patched
         `fit_generator()` method calling a patched `fit()` method, we only patch
         `fit_generator()` in Keras < 2.4.0.
         """
-        original = gorilla.get_original_attribute(keras.Model, "fit_generator")
         unlogged_params = ["self", "generator", "callbacks", "validation_data", "verbose"]
         return _run_and_log_function(self, original, args, kwargs, unlogged_params, 4)
 
-    wrap_patch(keras.Model, "fit", fit)
+    safe_patch(FLAVOR_NAME, keras.Model, "fit", fit, manage_run=True)
```

Comment:
TODO (@dbczumar): Use `try_mlflow_log` in place of certain try/except statements in Keras autologging